packages/lib/src/components/Card/components/CardInput/CardInput.tsx

```diff
@@ -1,47 +1,43 @@
 import SecuredFieldsProvider, { SFPState } from '../../../internal/SecuredFields/SecuredFieldsProvider';
 import defaultProps from './defaultProps';
 import defaultStyles from './defaultStyles';
-import styles from './CardInput.module.scss';
 import './CardInput.scss';
-import LoadingWrapper from '../../../internal/LoadingWrapper';
-import StoredCardFields from './components/StoredCardFields';
-import Installments from './components/Installments';
-import CardFields from './components/CardFields';
-import KCPAuthentication from './components/KCPAuthentication';
-import SocialSecurityNumberBrazil from '../../../internal/SocialSecurityNumberBrazil/SocialSecurityNumberBrazil';
-import StoreDetails from '../../../internal/StoreDetails';
-import Address from '../../../internal/Address/Address';
-import getImage from '../../../../utils/get-image';
-import { CardInputProps, CardInputValidState, CardInputErrorState, CardInputDataState } from './types';
-import { ALL_SECURED_FIELDS, CVC_POLICY_REQUIRED, DATE_POLICY_REQUIRED } from '../../../internal/SecuredFields/lib/configuration/constants';
+import { CardInputProps, CardInputValidState, CardInputErrorState, CardInputDataState, CardInputRef } from './types';
+import { CVC_POLICY_REQUIRED, DATE_POLICY_REQUIRED } from '../../../internal/SecuredFields/lib/configuration/constants';
 import { BinLookupResponse } from '../../types';
 import { cardInputFormatters, cardInputValidationRules, getRuleByNameAndMode } from './validate';
 import CIExtensions from '../../../internal/SecuredFields/binLookup/extensions';
-import { CbObjOnFocus } from '../../../internal/SecuredFields/lib/types';
-import CardHolderName from './components/CardHolderName';
 import useForm from '../../../../utils/useForm';
-import { ErrorPanel, ErrorPanelObj } from '../../../../core/Errors/ErrorPanel';
+import { ErrorPanelObj } from '../../../../core/Errors/ErrorPanel';
 import { getLayout, sortErrorsForPanel } from './utils';
-import { selectOne } from '../../../internal/SecuredFields/lib/utilities/dom';
 import { AddressData } from '../../../../types';
 import Specifications from '../../../internal/Address/Specifications';
 import { ValidationRuleResult } from '../../../../utils/Validator/Validator';
+import { StoredCardFieldsWrapper } from './components/StoredCardFieldsWrapper';
+import { CardFieldsWrapper } from './components/CardFieldsWrapper';
+import getImage from '../../../../utils/get-image';
+import styles from './CardInput.module.scss';
+import { getErrorPanelHandler, getAddressHandler, getFocusHandler } from './handlers';
+import { InstallmentsObj } from './components/Installments/Installments';
 
-function CardInput(props: CardInputProps) {
+const CardInput: FunctionalComponent<CardInputProps> = props => {
     const sfp = useRef(null);
     const billingAddressRef = useRef(null);
     const isValidating = useRef(false);
 
+    const cardInputRef = useRef<CardInputRef>({});
+    props.setComponentRef(cardInputRef.current);
```

Comment:
Is it intentional to leave this `props.setComponentRef(cardInputRef.current);` outside of an effect? As it is right now, I expect it to set the component ref multiple times without need

Reply:
I wasn't quite sure what to do here. `useMemo` seemed like overkill for such a minor operation. So I've found another way to ensure this is only called once